### PR TITLE
feat: Remove deployment-RP warning suppression from `bicepconfig.json`

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

This PR should only be merged after
- #2350
- #2353 
- #2354 
- #2355 
- #2498 

## Type of Change

- [x] Update to CI Environment or utilities (Non-module affecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: bicepconfig.json [hunk 1]
```diff
@@ -6,11 +6,7 @@
     },
     "analyzers": {
         "core": {
-            "rules": {
-                "no-deployments-resources": {
-                    "level": "off" // Reason: The telemetry resource only has a single output and is self-contained in a deployment
-                }
-            }
+            "rules": {}
```
Comment (eriqua) [inline bicepconfig.json:9]: Can we have just one pipeline badge?

Comment (AlexanderSehr) [reply bicepconfig.json:9]: Sure. I wouldn't know why as these are Bicep Linting rules that don't matter to any of our logic, but let me find a very small module 😄

Comment (AlexanderSehr) [reply bicepconfig.json:9]: [avm.res.analysis-services.server](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.analysis-services.server.yml)

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**